lib/src/solver/incompatibility_cause.dart

```diff
@@ -81,6 +81,11 @@ class SdkCause extends IncompatibilityCause {
   /// The SDK with which the package was incompatible.
   final Sdk sdk;
 
+  bool get noNullSafetyCause =>
+      sdk.isDartSdk &&
+      !LanguageVersion.fromSdkConstraint(constraint).supportsNullSafety &&
+      sdk.version! >= Version(3, 0, 0).firstPreRelease;
```

Comment:
I wish we could make this private, or something, but I don't see a good way to do it....

So I suggest we do this :D